File: leetcode/problems/1-two-sum/1-two-sum.go
package main

func twoSum(nums []int, target int) []int {
	res := []int{-1, -1}
	if len(nums) < 2 {
		return res
	}

	hashMap := map[int]int{}
	other := 0
	for i := range nums {
		other = target - nums[i]
		if _, ok := hashMap[other]; ok {
			res[0] = hashMap[other]
			res[1] = i
		}

		hashMap[nums[i]] = i
	}

	return res
}

File: leetcode/problems/language-specific/python/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start

# * Hash Table Solution | O(n) Time | O(n) Space
# * n -> The length of nums array

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_mappings = {}
        for idx, num in enumerate(nums):
            other_val = target - num
            if other_val in index_mappings:
                return [index_mappings[other_val], idx]

            index_mappings[num] = idx

# @lc code=end
